flowmachine/tests/test_subscriber_distance_series.py

```diff
@@ -0,0 +1,152 @@
+    Test that we get expected return values for the various statistics
+    """
+    sub_a_id, sub_b_id = "j6QYNbMJgAwlVORP", "NG1km5NzBg5JD8nj"
+    rl = daily_location("2016-01-01", spatial_unit=make_spatial_unit("lon-lat"))
+    df = get_dataframe(
+        DistanceSeries(
+            subscriber_locations=SubscriberLocations(
+                "2016-01-01", "2016-01-07", spatial_unit=make_spatial_unit("lon-lat")
+            ),
+            reference_location=rl,
+            statistic=stat,
+        )
+    ).set_index(["subscriber", "datetime"])
+    assert df.loc[(sub_a_id, date(2016, 1, 1))].value == pytest.approx(sub_a_expected)
+    assert df.loc[(sub_b_id, date(2016, 1, 6))].value == pytest.approx(sub_b_expected)
+
+
+@pytest.mark.parametrize(
+    "stat, sub_a_expected, sub_b_expected",
+    [
+        ("min", 9284030.27181416, 9123237.1676943),
+        ("median", 9327090.49789517, 9348965.25483016),
+        ("stddev", 213651.966918163, 180809.405030127),
+        ("avg", 9428284.48898054, 9390833.62618702),
+        ("sum", 28284853.4669416, 112690003.514244),
+        ("variance", 45647162968.0, 32692040947.3485),
+    ],
+)
+def test_returns_expected_values_fixed_point(
+    stat, sub_a_expected, sub_b_expected, get_dataframe
+):
+    """
+    Test that we get expected return values for the various statistics with 0, 0 reference
+    """
+    sub_a_id, sub_b_id = "j6QYNbMJgAwlVORP", "NG1km5NzBg5JD8nj"
+    df = get_dataframe(
+        DistanceSeries(
+            subscriber_locations=SubscriberLocations(
+                "2016-01-01", "2016-01-07", spatial_unit=make_spatial_unit("lon-lat")
+            ),
+            statistic=stat,
+        )
+    ).set_index(["subscriber", "datetime"])
+    assert df.loc[(sub_a_id, date(2016, 1, 1))].value == pytest.approx(sub_a_expected)
+    assert df.loc[(sub_b_id, date(2016, 1, 6))].value == pytest.approx(sub_b_expected)
+
+
+def test_no_cast_for_below_day(get_dataframe):
+    """
+    Test that results aren't cast to date for smaller time buckets.
+    """
+    df = get_dataframe(
+        DistanceSeries(
+            subscriber_locations=SubscriberLocations(
+                "2016-01-01", "2016-01-02", spatial_unit=make_spatial_unit("lon-lat")
+            ),
+            time_bucket="hour",
+        )
+    )
+    assert isinstance(df.datetime[0], datetime)
+
+
+def test_error_when_subs_locations_not_point_geom():
+    """
+    Test that error is raised if the spatial unit of the subscriber locations isn't point.
+    """
+
+    with pytest.raises(ValueError):
+        DistanceSeries(
+            subscriber_locations=SubscriberLocations(
+                "2016-01-01", "2016-01-07", spatial_unit=make_spatial_unit("admin3")
+            )
+        )
+
+
+def test_error_on_spatial_unit_mismatch():
+    """
+    Test that error is raised if the spatial unit of the subscriber locations isn't point.
+    """
+
+    rl = daily_location("2016-01-01")
+
+    with pytest.raises(ValueError):
```

Comment:
Would be good to include the `matches=...` argument in any calls to `pytest.raises` to ensure that the errors are raised for the expected reasons. (Every now and then I hit some code that raises the expected exception type but for a different reason than I thought.)

E.g.:
```suggestion
    with pytest.raises(ValueError, match="reference_location must have the same spatial unit as subscriber_locations"):
```

(Also in the other tests below.)